cmd/imagedigestexporter/main.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var (
+	images = flag.String("images", "", "List of images resources built by task in json format")
+)
+
+func main() {
+	flag.Parse()
+
+	imageResources := []*v1alpha1.ImageResource{}
+	err := json.Unmarshal([]byte(*images), &imageResources)
+	if err != nil {
+		log.Fatalf("Error reading images array: %v", err)
+	}
+
+	output := []v1alpha1.PipelineResourceResult{}
+	for _, imageResource := range imageResources {
+		ii, err := layout.ImageIndexFromPath(imageResource.IndexPath)
+		if err != nil {
+			// if this image doesn't have a builder that supports index.josn file,
+			// then it will be skipped
```

Comment:
bumping this! :)

Reply:
bumping this! :)

Reply:
I don't want to log something here because the controller will be trying to parse the output of that container, also because not all builder will support this, I want it to be optional without errors.

Reply:
hm okay :( too bad it [looks like](https://github.com/tektoncd/pipeline/blob/51a92ab71ce96140fef8ea74651b9dcd22450db2/vendor/k8s.io/api/core/v1/types.go#L4278) you couldn't have the controller only grab only stdout :(

Reply:
Let me give that a try :)

Reply:
@bobcatfish as far as I understand you have to attach to the pod to get the `Stdout` only, which would not work if the pod is completed. I think the get logs would be a safer bet in that case, wdyt?

Reply:
I understand the issue with using log here, however silently skipping the image if the builder is not supported will make things easier to troubleshoot. You may also have the case of a supported image builder which is malfunctioning in a specific case and not producing the index.json.
 
Would it be an option to use the digest field itself to output a standard string e.g. "builder did not produce index.json" or so?

Reply:
Maybe the controller could be aware of that error and not try to parse if it sees the error?